src/bound.jl

```diff
@@ -10,8 +10,31 @@ const IntervalBound = Tuple{Float64,Float64}
 """
 $(TYPEDEF)
 
-Shortcut for possible bounds: either no bound is present (`nothing`), or a
-single number is interpreted as an exact equality bound, or a tuple of 2
-numbers is interpreted as an interval bound.
+A special type of bound where the variable may only take on the values 0 or 1
+exclusively.
 """
-const Bound = Union{Nothing,Float64,IntervalBound}
+struct BinaryBound end
+
+const Binary = BinaryBound()
+
+"""
+$(TYPEDEF)
+
+A special type of bound where the variable may only take on integer values.
+"""
+struct IntegerBound end
```

Comment:
does JuMP have any support for limiting the integers to a given interval? or better have that as a separate constraint?

Reply:
I think you just bound then later in constraints